docs/src/components/ComponentDoc/ComponentExample/ComponentExample.js

```diff
@@ -64,36 +62,44 @@ class ComponentExample extends PureComponent {
     renderHtml: true,
   }
 
-  componentWillMount() {
-    const { examplePath } = this.props
-    this.anchorName = examplePathToHash(examplePath)
-
-    this.setState({
-      showCode: this.isActiveHash(),
-      sourceCode: this.getOriginalSourceCode(),
-    })
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const { examplePath, exampleSources, location } = nextProps
-    const nextSourceCode = exampleSources[examplePath]
+  static getDerivedStateFromProps(nextProps, prevState = {}) {
+    const {
+      exampleKeys,
+      examplePath,
+      exampleSources,
+      location: { hash },
+    } = nextProps
+
+    const anchorName = examplePathToHash(examplePath)
```

Comment:
We can move `anchorName` to `constructor()` as we need it only once

Reply:
We can do that only in case if `examplePath` property is never changed. Current usage of `ComponentExample` implies such usage. But it is a little bit confusing from general perspective.

Reply:
Based on nature of `ComponentExample` it's impossible in production and almost impossible in development, so actually we can do this :)

Reply:
Ok. I got it. Another reason to have `anchorName` in state is the ability to use it in static context during derived state computation. If the point for having it as a class field is to perform a computation just once I can do a conditional assignment to the non-empty anchorName.